asynciterator.ts

```diff
@@ -1852,17 +1894,166 @@ class HistoryReader<T> {
   }
 }
 
+/**
+ * The IteratorIterator class provides a performant wrapper
+ * to convert ES Iterable instances into AsyncIterator instances.
+ */
+export class IteratorIterator<T> extends AsyncIterator<T> {
+  private _source: Iterator<T>;
+
+  /**
+   *
+   * @param source
+   */
+  constructor(source: Iterator<T>) {
+    super();
+    this._source = source;
+    this.readable = true;
+  }
+
+  read(): T | null {
+    let value: T | null = null;
+    if (this._source) {
+      const item = this._source.next();
+      if (!item.done)
+        value = item.value;
+      else
+        this.close();
+    }
+    return value;
+  }
```

Comment:
So as to avoid unecessary variable assignment, and to avoid branching on the common case, why not

```ts
if (!this._source)
  return null
const item = this._source.next();
if (item.done) {
   this.close();
   return null;
}
return item.value
```

Reply:
Done!